architecture/2024-02-application-model-promises.md

```diff
@@ -0,0 +1,173 @@
+  - Corollary: The application model solves the *lowest-common-denominator* problem by giving users ways to punch through the abstraction when desired.
+- Application definitions should be minimally coupled to operational concerns. We leverage *separation-of-concerns* to avoid mixing application and operational configuration.
+
+### User scenarios (optional)
+
+These user scenarios are somewhat abstract but help provide motivation for why this discussion is valuable.
+
+#### Managing Radius Environments
+
+Alfonso manages environment definitions for multiple clouds and runtimes as part of a platform engineering team. He writes Terraform templates that create the supporting infrastructure like AKS, EKS, or ECS clusters. This includes supporting infrastructure like secret stores or diagnostics dashboards. He also includes the relevant recipes for services like AWS MemoryDB or Azure Cache for Redis that the platform team supports. 
+
+These templates are parameterized and used by developers as part of a self-service environment creation workflow. Because of the promises defined by Radius, Alfonso knows these environments will meet the organization's requirements.
+
+#### Deploying Radius Applications
+
+Janine builds business applications using Java, and deploys them with Radius. She's responsible for providing the application definition using Radius concepts like containers, gateways, and recipes. 
+
+These templates are deployed through a CI/CD process to multiple kinds of environments. Because of the promises provided by Radius and the responsibilities fulfiled by Alfonso, the application definition can be deployed to any supported platform.
+
+## Design
+
+### Introducing concepts
+
+*These are the core concepts we'll be working with in this document. Where it's helpful we generalize to simplify the discussion.*
+
+---
+
+**Environment:** A pool of compute, networking, and supporting resources use by potentially many applications. Applications inside the environment share the pool of available compute and exist inside the same network namespace. The *supporting resources* address potentially *many* different needs and scenarios.
+
+For our purposes, a supporting resource of the environment probably has:
+
+- The same lifecycle as the environment.
+- Is shared for all applications in the environment (not part of a specific application).
+- Is likely not managed by developers.
+- Likely has some direct integration with Radius:
+  - Storing secrets.
+  - Storing diagnostic data.
+
+---
+
+**Application:** A composite grouping of compute and non-compute resources defined by the developer. The application is explictly not tied to access control or to deployment/lifecycle. The application is a boundary for common configuration and metadata. 
+
+The question is often asked, "what does an application *mean*?" Our answer is that the application is whatever the developers say it is.
+
+**Compute Resource:**
+
+
```

Comment:
Do we need to add definition for groups/resourcegroups ??

Reply:
+1. And how is the resource group related to an env